K epsilon model

tanoret opened the pull request:
Refs #9007

tanoret committed:
correcting diffusion model Refs #9007

tanoret committed:
adding k-epsilon source and sink Refs #9007

tanoret committed:
relaxing k-epsilon Refs #9007

tanoret committed:
adding wall treatement Refs #9007

tanoret committed:
making explicit k-epsilon Refs #9007

tanoret committed:
adding mu_t wall functions Refs #9007

tanoret committed:
testing and inlet conditions Refs #9007

tanoret committed:
adding temperature wall functions Refs #9007

tanoret committed:
adding lid-driven example Refs #9007

Commit:
msfr test Refs #9007

tanoret committed:
ERCOFTAC BFS Refs #9007

tanoret committed:
adding tests Refs #9007

tanoret committed:
adding custom marking methods Refs #9007

tanoret committed:
amending source sink term Refs #9007

tanoret committed:
adding docs Refs #9007

tanoret commented:
@grmnptr Could you also take a look please? Thanks!

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1898945/) on 8245d1d wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format ec85e86826fda2e21ce63bb3a8594ec63bdb07d7`

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1899398/) on d50487a wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/25752/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1899439/) on d50487a wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

### Navier stokes

<table>
  <thead>
    <tr>
      <th></th>
      <th>ec85e8</th>
      <th colspan=4>#25752 d50487</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>84.87%</td>
      <td>85.04%</td>
      <td>+0.16%</td>
      <td>87.92%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>12574</td>
      <td>13366</td>
      <td>+792</td>
      <td>808</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>2241</td>
      <td>2352</td>
      <td>+111</td>
      <td>111</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/25752/coverage_diff/navier_stokes)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/25752/coverage/navier_stokes)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/25752/coverage/xfem)

</details>

## Warnings

- `navier_stokes` new line coverage rate 87.92% is less than the suggested 90.0%



This comment will be updated on new commits.

grmnptr reviewed (changes requested):
Submitting after the documentation. Vast majority of the comments are typos so nothing serious. There are some style changes (like the header structure) but feel free to disregard those. Appreciated the attention to details when it came to describing different effects!

tanoret committed:
adding missing includes and docs Refs #9007

tanoret commented on modules/navier_stokes/doc/content/source/auxkernels/kEpsilonViscosityAux.md:
```diff
@@ -0,0 +1,34 @@
+- $k$ is the turbulent kinetic energy,
+- $\epsilon$ is the turbulent kinetic energy dissipation rate.
+
+Via the parameter `bulk_wall_treatment = true` the methods allows us to set the value
```
Thanks

tanoret commented on modules/navier_stokes/doc/content/source/auxkernels/kEpsilonViscosityAux.md:
```diff
@@ -0,0 +1,34 @@
+- $\epsilon$ is the turbulent kinetic energy dissipation rate.
+
+Via the parameter `bulk_wall_treatment = true` the methods allows us to set the value
+of the cells identified in contact with the boundaries identified in `walls` to the dynamic
```
Thanks

tanoret commented on modules/navier_stokes/doc/content/source/auxkernels/kEpsilonViscosityAux.md:
```diff
@@ -0,0 +1,34 @@
+
+!alert note
+If the boundary conditions for the dynamic turbulent viscosity are already set via [INSFVTurbulentViscosityWallFunction](INSFVTurbulentViscosityWallFunction.md),
+there is no need to add bulk wall treatment, i.e., we should set `bulk_wall_treatment = false`.
```
Thanks

tanoret commented on modules/navier_stokes/doc/content/source/auxkernels/kEpsilonViscosityAux.md:
```diff
@@ -0,0 +1,34 @@
+!alert note
+If the boundary conditions for the dynamic turbulent viscosity are already set via [INSFVTurbulentViscosityWallFunction](INSFVTurbulentViscosityWallFunction.md),
+there is no need to add bulk wall treatment, i.e., we should set `bulk_wall_treatment = false`.
+This type of bulk wall treatment is mainly design for porous media formulations
```
Thx!

tanoret commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVInletIntensityTKEBC.md:
```diff
@@ -0,0 +1,24 @@
+
+where:
+
+- I is the turbulent intensity, which can be set by the user or computed via correlations, e.g., $I = 0.16 Re^{-\frac{1}{8}}$
```
Thanks!

tanoret commented on modules/navier_stokes/doc/content/source/auxkernels/TurbulentConductivityAux.md:
```diff
@@ -0,0 +1,21 @@
+# TurbulentConductivityAux
+
+!syntax description /AuxKernels/TurbulentConductivityAux
+
```
Thanks!

grmnptr replied:
Could you do this for the other doc files too, thanks!

tanoret commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVTurbulentTemperatureWallFunction.md:
```diff
@@ -0,0 +1,88 @@
+
+- Sub-laminar region: $y^+ \le 5$
+- Buffer region: $y^+ \in (5, 30)$
+- Logarithmic region: $y^+ \ge 5$
```
Nice catch! Thx!

tanoret commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVTurbulentTemperatureWallFunction.md:
```diff
@@ -0,0 +1,88 @@
+- Buffer region: $y^+ \in (5, 30)$
+- Logarithmic region: $y^+ \ge 5$
+
+For the procedures for determining the non-dimensional wall distance as $y^+$
```
Thank s

tanoret commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVTurbulentTemperatureWallFunction.md:
```diff
@@ -0,0 +1,88 @@
+- $u_{\tau}$ is the friction velocity defined by law of the wall
+- $y_p$ is the distance from the boundary to the centroid of the near-wall cell
+- $Pr_t$ is the turbulent Prandtl number, which typically ranges between 0.3 and 0.9
+- $w_s$ is a near-wall scaling factor that is defined  as follows:
```
Tx

tanoret commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVTurbulentTemperatureWallFunction.md:
```diff
@@ -0,0 +1,88 @@
+
+- $\kappa = 0.4187$ is the von Kármán constant
+- $E = 9.793$ is a closure parameter
+- $J_k$ is the Jayatilleke wall functions defined as follows:
```
Yes. Agree. Are you good with trying to get this merged and then improve the docs?

grmnptr replied:
Sure, if you pinky-promise that you will do it.

tanoret commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVTurbulentTemperatureWallFunction.md:
```diff
@@ -0,0 +1,88 @@
+is defined via a linear blending function as follows:
+
+\begin{equation}
+  \alpha_b = \alpha_t \frac{(y^+ - 5)}{5} + \alpha_l \left[ 1 - \frac{(y^+ - 5)}{5} \right]
```
Nice catch. Thanks!

tanoret commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVTurbulentTemperatureWallFunction.md:
```diff
@@ -0,0 +1,88 @@
+  \alpha_b = \alpha_t \frac{(y^+ - 5)}{5} + \alpha_l \left[ 1 - \frac{(y^+ - 5)}{5} \right]
+\end{equation}
+
+Finally, using the thermal diffusivity, the heat flux as the wall is defined
```
Tx

tanoret commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVTurbulentTemperatureWallFunction.md:
```diff
@@ -0,0 +1,88 @@
+where:
+
+- $k$ is the thermal conductivity
+- $rho$ is the density
```
Tx

tanoret commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVTurbulentViscosityWallFunction.md:
```diff
@@ -0,0 +1,146 @@
+
+!syntax description /FVBCs/INSFVTurbulentViscosityWallFunction
+
+Implements wall functions boundary conditions for the turbulent
```
Tx!

tanoret commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVTurbulentViscosityWallFunction.md:
```diff
@@ -0,0 +1,146 @@
+
+- Sub-laminar region: $y^+ \le 5$
+- Buffer region: $y^+ \in (5, 30)$
+- Logarithmic region: $y^+ \ge 5$
```
Thanks!

tanoret commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVTurbulentViscosityWallFunction.md:
```diff
@@ -0,0 +1,146 @@
+- Logarithmic region: $y^+ \ge 5$
+
+Four different formulations are supported
+as defined by the `wall_treatment` parameter.
```
Tx!

tanoret commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVTurbulentViscosityWallFunction.md:
```diff
@@ -0,0 +1,146 @@
+Four different formulations are supported
+as defined by the `wall_treatment` parameter.
+
+*Equilibrium wall functions using a Newton solve*
```
Tx!

tanoret commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVTurbulentViscosityWallFunction.md:
```diff
@@ -0,0 +1,146 @@
+
+*Equilibrium wall functions using a Newton solve*
+
+These treatment is enabled by the parameter `wall_treatment = eq_newton`.
```
Thanks!

tanoret commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVTurbulentViscosityWallFunction.md:
```diff
@@ -0,0 +1,146 @@
+*Equilibrium wall functions using a Newton solve*
+
+These treatment is enabled by the parameter `wall_treatment = eq_newton`.
+The treatment uses equilibrium wall functions where the following formulatiopn is used
```
Thx!

tanoret commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVTurbulentViscosityWallFunction.md:
```diff
@@ -0,0 +1,146 @@
+- $\kappa = 0.4187$ is the von Kármán constant
+
+In this method, we iterate on the wall function and $y^+$ to find
+$u_{\tau}$ as via a Newton solve. Once $u_{\tau}$ defined, $y^+$ is
```
Thanks!

tanoret commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVTurbulentViscosityWallFunction.md:
```diff
@@ -0,0 +1,146 @@
+- $\kappa = 0.4187$ is the von Kármán constant
+
+In this method, we iterate on the wall function and $y^+$ to find
+$u_{\tau}$ as via a Newton solve. Once $u_{\tau}$ defined, $y^+$ is
```
Tx!

tanoret commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVTurbulentViscosityWallFunction.md:
```diff
@@ -0,0 +1,146 @@
+
+In this method, we iterate on the wall function and $y^+$ to find
+$u_{\tau}$ as via a Newton solve. Once $u_{\tau}$ defined, $y^+$ is
+computed defining the boundary turbulent viscosity.
```
Thanks!

tanoret commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVTurbulentViscosityWallFunction.md:
```diff
@@ -0,0 +1,146 @@
+contains nodes in the buffer layer is not recommended.
+
+
+*Equilibrium wall functions using a fixed-point solve*
```
Thanks!

tanoret commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVTurbulentViscosityWallFunction.md:
```diff
@@ -0,0 +1,146 @@
+
+*Equilibrium wall functions using a fixed-point solve*
+
+These treatment is enabled by the parameter `wall_treatment = eq_incremental`.
```
Thanks!

tanoret commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVTurbulentViscosityWallFunction.md:
```diff
@@ -0,0 +1,146 @@
+
+*Equilibrium wall functions using linearized wall function*
+
+These treatment is enabled by the parameter `wall_treatment = eq_linearized`.
```
Thanks!

tanoret commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVTurbulentViscosityWallFunction.md:
```diff
@@ -0,0 +1,146 @@
+Then, $y^+$ is computed from $u_{\tau}$.
+
+!alert note
+`eq_linearized` will work fast as there is no nonlienar solve at
```
Tx!

tanoret commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVTurbulentViscosityWallFunction.md:
```diff
@@ -0,0 +1,146 @@
+the near-wall region. However, the method may introduce significant
+near-wall errors. The method is designed to be used in conjunction
+with porous media treatement and not necessairly for free flow.
+
```
Tx

tanoret commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVTurbulentViscosityWallFunction.md:
```diff
@@ -0,0 +1,146 @@
+with porous media treatement and not necessairly for free flow.
+
+
+*Non-equilibrium wall functions*
```
Tx

tanoret commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVTurbulentViscosityWallFunction.md:
```diff
@@ -0,0 +1,146 @@
+
+*Non-equilibrium wall functions*
+
+These treatment is enabled by the parameter `wall_treatment = neq`.
```
Tanks!

tanoret commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKEDSourceSink.md:
```diff
@@ -0,0 +1,114 @@
+
+!syntax description /FVKernels/INSFVTKEDSourceSink
+
+The object computes the turbulent source and sink term for the turbulent kinetic energy equation.
```
Thanks!

tanoret commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKEDSourceSink.md:
```diff
@@ -0,0 +1,114 @@
+
+The object computes the turbulent source and sink term for the turbulent kinetic energy equation.
+
+Two terms are computed `destruction` and `production` and the term `destruction - production` is
```
Thanks

tanoret commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVMomentumDiffusion.md:
```diff
@@ -9,6 +9,17 @@ Navier-Stokes momentum equation, e.g.
 
 where $\mu$ is the dynamic viscosity, and $\bm{v}$ is the velocity.
 
+The object also takes a parameter `complete_expansion`, which is
```
Thanks!

tanoret commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVMomentumDiffusion.md:
```diff
@@ -9,6 +9,17 @@ Navier-Stokes momentum equation, e.g.
+\end{equation}
+
+!alert note
+The term $\nabla \cdot \mu (\nabla \bm{v})^T = 0$ for incompressible flow.
```
Thanks

tanoret commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKEDSourceSink.md:
```diff
@@ -0,0 +1,114 @@
+passed to the residual.
+A different treatment is used for the bulk and the near wall regions.
+
+*Bulk formulation*:
```
Thanks!

tanoret commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKEDSourceSink.md:
```diff
@@ -0,0 +1,114 @@
+- $S$ is the shear strain tensor internal norm, defined as $S = \sqrt{2\mathbf{S}:\mathbf{S}}$ with the shear strain tensor defined as $\mathbf{S} = \frac{1}{2} [\nabla \vec{u} + (\nabla \vec{u})^T]$,
+- $k$ is the turbulent kinetic energy, which generally comes from a coupled transport equation or can be set by the user for reproducing canonical cases.
+
+The dissipation rate of the turbulent kinetic energy dissipation rate is modeled as follows:
```
Thanks!

tanoret commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKEDSourceSink.md:
```diff
@@ -0,0 +1,114 @@
+where:
+
+- $C_{2,\epsilon} = 1.92$ is a closure parameter,
+- $\epsilon$ is the solution variable, i.e., the dissipation rate of the turbulent kinetic energy dissipation rate,
```
Thanks

tanoret commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKEDSourceSink.md:
```diff
@@ -0,0 +1,114 @@
+
+- $C_{2,\epsilon} = 1.92$ is a closure parameter,
+- $\epsilon$ is the solution variable, i.e., the dissipation rate of the turbulent kinetic energy dissipation rate,
+- $t_k = \frac{k}{\epsilon}$ is the turbulent time scale; if the `linearized_model = true`, this timescale is computed from the previous iteration; if `linearized_model = false`, in a nonlinear solve, this timescale is aded to the Jacobian.
```
Thanks

tanoret commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKEDSourceSink.md:
```diff
@@ -0,0 +1,114 @@
+- $\epsilon$ is the solution variable, i.e., the dissipation rate of the turbulent kinetic energy dissipation rate,
+- $t_k = \frac{k}{\epsilon}$ is the turbulent time scale; if the `linearized_model = true`, this timescale is computed from the previous iteration; if `linearized_model = false`, in a nonlinear solve, this timescale is aded to the Jacobian.
+
+*Wall formulation*:
```
Thanks

tanoret commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKEDSourceSink.md:
```diff
@@ -0,0 +1,114 @@
+A separate formulation is used for the `sub-laminar` and `logarithmic` boundary layers.
+The determination of whether the near-wall cell lies in the laminar or logarithmic region
+is performed via the non-dimensional wall distance $y^+$.
+The non-dimensional wall distance is defined as defined differently according to the
```
Thanks

tanoret commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKEDSourceSink.md:
```diff
@@ -0,0 +1,114 @@
+The non-dimensional wall distance is defined as defined differently according to the
+`non_equilibrium_treatement`.
+
+If `non_equilibrium_treatement = false`, the standard wall function formulations is used in
```
Thanks

tanoret commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKEDSourceSink.md:
```diff
@@ -0,0 +1,114 @@
+`non_equilibrium_treatement`.
+
+If `non_equilibrium_treatement = false`, the standard wall function formulations is used in
+which the $y^+$ is found by an incremental fixed point algorithm as follows:
```
Tx

tanoret commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKEDSourceSink.md:
```diff
@@ -0,0 +1,114 @@
+- $u_{\tau}$ is the friction velocity, defined as $u_{\tau} = \sqrt{\frac{\tau_w}{\rho}}$ with $\tau_w$ the shear stress at the wall for which the condition is applied,
+- $\mu$ is the dynamic molecular viscosity.
+
+If `non_equilibrium_treatement = true`, non-equilibrium wall function formulations is
```
Tx

tanoret commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKEDSourceSink.md:
```diff
@@ -0,0 +1,114 @@
+\end{equation}
+
+!alert note
+Using non-equilibrium wall functions is recomended for problems with recirculations and boundary layer detachment. However, using non-equilibrium wall functions will deteriorate results for standard problems such as flow developing over walls.
```
Grazie!

tanoret commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKEDSourceSink.md:
```diff
@@ -0,0 +1,114 @@
+!alert note
+Using non-equilibrium wall functions is recomended for problems with recirculations and boundary layer detachment. However, using non-equilibrium wall functions will deteriorate results for standard problems such as flow developing over walls.
+
+The cells belonging to the `sub-laminar` boundary layers are defined as those
+for which $y^+ < 11.25$.
```
Danke

tanoret commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKEDSourceSink.md:
```diff
@@ -0,0 +1,114 @@
+
+where:
+
+- $\kappa = 0.41$ is the von Kármán constant.
```
Yes, it matters. Thanks!

tanoret commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKEDSourceSink.md:
```diff
@@ -0,0 +1,114 @@
+- $\kappa = 0.41$ is the von Kármán constant.
+
+!alert note
+When using wall functions, since the equilibrium value for $\epsilon$ is set in the cells near the wall, the user is recomended to deactivate advection and diffusion for those near wall cells.
```
The user can do it. It is about the "correct" utilization of the model. Thanks!

tanoret commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKEDSourceSink.md:
```diff
@@ -0,0 +1,114 @@
+
+!alert note
+When using near-wall treatment, we assume that the $\mu_t$ funtor is computed by an object
+that performs near-wall treatment. Otherwise, the results obtained won't not physically correct
```
Thanks!

tanoret commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKESourceSink.md:
```diff
@@ -0,0 +1,109 @@
+passed to the residual.
+A different treatment is used for the bulk and the near wall regions.
+
+*Bulk formulation*:
```
Thanks

tanoret commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKESourceSink.md:
```diff
@@ -0,0 +1,109 @@
+However, for canonical or meassured cases, e.g., isotropic decaying turbulece,
+the user can force its value to a functor.
+
+*Wall formulation*:
```
Thanks

tanoret commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKESourceSink.md:
```diff
@@ -0,0 +1,109 @@
+
+All cells in contact with a boundary identified in the `walls` list are applied a different
+treatment for production and destruction.
+A separate formulation is used for the `sub-laminar` and `logarithmic` boundary layers.
```
Tx

tanoret commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKESourceSink.md:
```diff
@@ -0,0 +1,109 @@
+
+- $k_{old}$ is the value of the turbulent kinetic energy in the previous iteration.
+
+Whereas the linear or nonlinear formulation is used is controled by the `linearized_model` parameter.
```
Thanks

tanoret commented on modules/navier_stokes/doc/content/source/materials/INSFVkEpsilonViscosityMaterial.md:
```diff
@@ -0,0 +1,22 @@
+
+!syntax description /Materials/INSFVkEpsilonViscosityMaterial
+
+This is the material class used to compute the dynamic turbulence viscosity
```
Thanks

tanoret committed:
addressing grmnptr comments Refs #9007

tanoret commented on modules/navier_stokes/doc/content/source/auxkernels/TurbulentConductivityAux.md:
```diff
@@ -0,0 +1,21 @@
+
+!syntax description /AuxKernels/TurbulentConductivityAux
+
+This is the axiliary kernel used to compute the thermal effective turbulence conductivity
```
Thanks. Unified to `turbulent viscosity`

Commit:
solving compiler warnings Refs #9007

tanoret committed:
fixing docs Refs #9007

Commit:
correcting non-unity build Refs #9007

grmnptr commented:
Ok, I am reviewing the code now.

grmnptr commented on modules/navier_stokes/include/auxkernels/kEpsilonViscosityAux.h:
```diff
@@ -0,0 +1,88 @@
+  virtual Real computeValue() override;
+
+  // Local method to find friction velocty
+  // Note: this method may be to need reimplemented for each new turbulent model
```
```suggestion
  /// This method may need to be reimplemented for each new turbulence model
```

grmnptr commented on modules/navier_stokes/include/auxkernels/kEpsilonViscosityAux.h:
```diff
@@ -0,0 +1,88 @@
+  // Note: this method may be to need reimplemented for each new turbulent model
+  ADReal findUStarLocalMethod(const ADReal & u, const Real & dist);
+
+  /// the dimension of the simulation
```
```suggestion
  /// The dimension of the domain
```

grmnptr commented on modules/navier_stokes/include/auxkernels/kEpsilonViscosityAux.h:
```diff
@@ -0,0 +1,88 @@
+  /// Wall boundaries
+  std::vector<BoundaryName> _wall_boundary_names;
+
+  /// Linearzied computation of y_plus ?
```
```suggestion
  /// If the user wants the linearized computation of y_plus
```

grmnptr commented on modules/navier_stokes/include/auxkernels/kEpsilonViscosityAux.h:
```diff
@@ -0,0 +1,88 @@
+  /// Linearzied computation of y_plus ?
+  const bool _linearized_yplus;
+
+  /// Linearzied computation of y_plus ?
```
```suggestion
  /// If the user wants to enable bulk wall treatment
```

grmnptr commented on modules/navier_stokes/include/auxkernels/kEpsilonViscosityAux.h:
```diff
@@ -0,0 +1,88 @@
+  /// Linearzied computation of y_plus ?
+  const bool _bulk_wall_treatment;
+
+  /// Non-equilibrium wall treatement ?
```
```suggestion
  /// IF the user requested non-equilibrium wall treatment
```

grmnptr commented on modules/navier_stokes/include/base/NavierStokesMethods.h:
```diff
@@ -99,10 +101,57 @@ Real prandtlPropertyDerivative(const Real & mu,
 ADReal findUStar(const ADReal & mu, const ADReal & rho, const ADReal & u, Real dist);
 
+/**
+ * Finds the nondimensional wall distance normalized with the friction velcoity
```
```suggestion
 * Finds the non-dimensional wall distance normalized with the friction velocity
```

grmnptr commented on modules/navier_stokes/include/fvbcs/INSFVMixingLengthTKEDBC.h:
```diff
@@ -0,0 +1,33 @@
+#include "FVDirichletBCBase.h"
+
+/**
+ * A class for velocity inlet boundary conditions
```
```suggestion
 * A class for turbulent kinetic energy dissipation rate inlet boundary conditions
```

grmnptr commented on modules/navier_stokes/include/fvbcs/INSFVTKEDWallFunctionBC.h:
```diff
@@ -0,0 +1,52 @@
+  const Moose::Functor<ADReal> & _rho;
+  /// Dynamic viscosity
+  const Moose::Functor<ADReal> & _mu;
+  /// Trubulent Dynamic viscosity
```
```suggestion
  /// Turbulent dynamic viscosity
```

grmnptr commented on modules/navier_stokes/include/fvbcs/INSFVTKEDWallFunctionBC.h:
```diff
@@ -0,0 +1,52 @@
+#include "FVFluxBC.h"
+
+/**
+ * Applies a Dirichlet boundary condition with a value prescribed by a function
```
```suggestion
 * Applies a wall function to the turbulent kinetic energy dissipation rate
```

grmnptr commented on modules/navier_stokes/include/fvbcs/INSFVTKEDWallFunctionBC.h:
```diff
@@ -0,0 +1,52 @@
+  ADReal boundaryValue(const FaceInfo & fi) const override;
+
+private:
+  /// the dimension of the simulation
```
```suggestion
  /// the dimension of the domain
```

grmnptr commented on modules/navier_stokes/include/fvbcs/INSFVTurbulentTemperatureWallFunction.h:
```diff
@@ -0,0 +1,57 @@
+#include "FVFluxBC.h"
+
+/**
+ * This boundary condition sets a constant heat flux with a splitting between the
+ * fluid and solid phases according to one of
```
```suggestion
 * This boundary condition applies a wall function for the energy equation for turbulent flows
```

grmnptr commented on modules/navier_stokes/include/fvbcs/INSFVTurbulentTemperatureWallFunction.h:
```diff
@@ -0,0 +1,57 @@
+protected:
+  virtual ADReal computeQpResidual() override;
+
+  /// the dimension of the simulation
```
```suggestion
  /// the dimension of the domain
```

grmnptr commented on modules/navier_stokes/include/fvbcs/INSFVTurbulentTemperatureWallFunction.h:
```diff
@@ -0,0 +1,57 @@
+  /// Linearized equation to find the wall function?
+  const bool _linearized_yplus;
+
+  /// Constant expressions
+  static constexpr Real _von_karman{0.4187};
+  static constexpr Real _E{9.793};
```
I think these could be lifted to `NS.h` looks like many kernels/bcs use it. So we can do `NS::von_karman` and `NS::E_turb` (if E is already taken) or something along those lines.

tanoret replied:
Copy-pasting from previous comment:
> This one is a bit tricky. In theory, they are constants. However, each term production, dissipation, wall functions, etc. that touches this constants could be used a slightly different value that works better for fitting. I'd say to keep it here for now

grmnptr commented on modules/navier_stokes/include/fvbcs/INSFVTurbulentViscosityWallFunction.h:
```diff
@@ -0,0 +1,55 @@
+#include "FVFluxBC.h"
+
+/**
+ * Applies a Dirichlet boundary condition with a value prescribed by a function
```
```suggestion
 * Applies a wall function to the turbulent viscosity field
```

grmnptr commented on modules/navier_stokes/include/fvbcs/INSFVTurbulentViscosityWallFunction.h:
```diff
@@ -0,0 +1,55 @@
+  const Moose::Functor<ADReal> & _rho;
+  /// Dynamic viscosity
+  const Moose::Functor<ADReal> & _mu;
+  /// Trubulent Dynamic viscosity
```
```suggestion
  ///  Turbulent dynamic viscosity
```

grmnptr commented on modules/navier_stokes/include/fvbcs/INSFVTurbulentViscosityWallFunction.h:
```diff
@@ -0,0 +1,55 @@
+  ADReal boundaryValue(const FaceInfo & fi) const override;
+
+private:
+  /// the dimension of the simulation
```
```suggestion
  /// the dimension of the domain
```

grmnptr commented on modules/navier_stokes/include/fvkernels/INSFVMomentumDiffusion.h:
```diff
@@ -41,4 +42,17 @@ class INSFVMomentumDiffusion : public INSFVFluxKernel
+  /// z-velocity
+  const INSFVVelocityVariable * const _w_var;
+
+  /// Boolean parameter to include the complete momentum exmapnsion
```
```suggestion
  /// Boolean parameter to include the complete momentum expansion
```

grmnptr commented on modules/navier_stokes/include/fvkernels/INSFVTKEDSourceSink.h:
```diff
@@ -0,0 +1,93 @@
+#include "INSFVVelocityVariable.h"
+
+/**
+ * Simple class to demonstrate off diagonal Jacobian contributions.
```
```suggestion
 * Computes the source and sink terms for the turbulent kinetic energy dissipation rate.
```

grmnptr commented on modules/navier_stokes/include/fvkernels/INSFVTKEDSourceSink.h:
```diff
@@ -0,0 +1,93 @@
+  const INSFVVelocityVariable * const _w_var;
+
+  /// Turbulent kinetic energy
+  // const INSFVVariable * const _k;
```
```suggestion
```

grmnptr commented on modules/navier_stokes/include/fvkernels/INSFVTKEDSourceSink.h:
```diff
@@ -0,0 +1,93 @@
+  /// Value of the first epsilon closure coefficient
+  const Real _C1_eps;
+
+  /// Value of the first epsilon closure coefficient
```
```suggestion
  /// Value of the second epsilon closure coefficient
```

grmnptr reviewed (changes requested):
Okay, I marked the comments that were addressed in the previous review resolved. Managed to finish the header files, posting it here. Will go on with the source files. 
So far so good!

tanoret commented on modules/navier_stokes/include/auxkernels/TurbulentConductivityAux.h:
```diff
@@ -0,0 +1,44 @@
+
+#include "INSFVVelocityVariable.h"
+/**
+ * Computes the turbuent viscosity for the k-Epsilon model.
```
Yes. It is more general as long as you have `mu_t` Thanks

tanoret commented on modules/navier_stokes/include/auxkernels/kEpsilonViscosityAux.h:
```diff
@@ -0,0 +1,88 @@
+  /// Relative tolerance to find the friction velocity
+  static constexpr Real _REL_TOLERANCE{1e-4};
+
+  /// -- Constants of the method
+  static constexpr Real _von_karman{0.4187};
+  static constexpr Real _E{9.793};
```
This one is a bit tricky. In theory, they are constants. However, each term production, dissipation, wall functions, etc. that touches this constants could be used a slightly different value that works better for fitting. I'd say to keep it here for now

tanoret commented on modules/navier_stokes/include/fvkernels/INSFVTKEDSourceSink.h:
```diff
@@ -0,0 +1,93 @@
+  const Real _C_mu;
+
+  /// Stored strain rate
+  std::map<const Elem *, Real> _symmetric_strain_tensor_norm_old;
```
Thanks

tanoret commented on modules/navier_stokes/include/fvkernels/INSFVTKEDSourceSink.h:
```diff
@@ -0,0 +1,93 @@
+  std::map<const Elem *, std::vector<Point>> _normal;
+  std::map<const Elem *, std::vector<const FaceInfo *>> _face_infos;
+
+  /// -- Constants of the method
+  static constexpr Real _von_karman{0.4187};
+  static constexpr Real E{9.793};
```
Copy pasting from previous comment:
> This one is a bit tricky. In theory, they are constants. However, each term production, dissipation, wall functions, etc. that touches this constants could be used a slightly different value that works better for fitting. I'd say to keep it here for now

tanoret commented on modules/navier_stokes/include/fvkernels/INSFVTKESourceSink.h:
```diff
@@ -0,0 +1,77 @@
+  std::map<const Elem *, std::vector<Point>> _normal;
+  std::map<const Elem *, std::vector<const FaceInfo *>> _face_infos;
+
+  /// -- Constants of the method
```
Copy-pasting from previous comments:
> This one is a bit tricky. In theory, they are constants. However, each term production, dissipation, wall functions, etc. that touches this constants could be used a slightly different value that works better for fitting. I'd say to keep it here for now

tanoret committed:
Apply suggestions from code review from grmnptr Refs #9007

Co-authored-by: Peter German <31662443+grmnptr@users.noreply.github.com>

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1919333/) on 7f655db wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 8cad10d94553be088dd177ac8edd3833a542c3b4`

tanoret committed:
removing extra examples Refs #9007

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1919382/) on 000ffaa wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 8cad10d94553be088dd177ac8edd3833a542c3b4`

grmnptr commented on modules/navier_stokes/src/fvkernels/INSFVTKEDSourceSink.C:
```diff
@@ -0,0 +1,253 @@
+      }
+    }
+
+    auto symmetric_strain_tensor_sq_norm =
+        (Sij_xx - trace) * grad_xx + Sij_xy * grad_xy + Sij_xz * grad_xz + Sij_xy * grad_yx +
+        (Sij_yy - trace) * grad_yy + Sij_yz * grad_yz + Sij_xz * grad_zx + Sij_yz * grad_zy +
+        (Sij_zz - trace) * grad_zz;
```
This could be done very easily, if you used tensors/vectors instead of each component separately. 
The tensor in libmesh has this contract function:

https://mooseframework.inl.gov/docs/doxygen/libmesh/classlibMesh_1_1TypeTensor.html#aca9325828337e17e35dfad38318f5664

Which will do the multiplication and summation for you. So you have one tensor for (A^T+A - trace*I) and you contract it with A.

tanoret replied:
Is this faster than doing a single component at the time? It was not in my experience...

tanoret replied:
The tensor implementation was my previous one and I had to code one component at the time because it was awfully slow

tanoret replied:
Maybe I am missing something you know that I don't

grmnptr commented on modules/navier_stokes/src/fvkernels/INSFVTKEDSourceSink.C:
```diff
@@ -0,0 +1,253 @@
+
+    production = _C1_eps * _rho(makeElemArg(_current_elem), state) * production_k;
+
+    auto time_scale = _k(makeElemArg(_current_elem), old_state) /
```
```suggestion
    const auto time_scale = _k(makeElemArg(_current_elem), old_state) /
```

grmnptr commented on modules/navier_stokes/src/fvkernels/INSFVTKESourceSink.C:
```diff
@@ -0,0 +1,267 @@
+  params.addParam<bool>(
+      "non_equilibrium_treatement",
+      false,
+      "Use non-equilibrium wall treatement (faster than standard wall treatement)");
```
```suggestion
      "Use non-equilibrium wall treatment (faster than standard wall treatment)");
```

grmnptr commented on modules/navier_stokes/src/fvkernels/INSFVTKEDSourceSink.C:
```diff
@@ -0,0 +1,253 @@
+      auto distance = _dist[_current_elem][i];
+
+      ADReal y_plus;
+      if (_non_equilibrium_treatement)
```
```suggestion
      if (_non_equilibrium_treatment)
```

grmnptr commented on modules/navier_stokes/src/fvkernels/INSFVTKESourceSink.C:
```diff
@@ -0,0 +1,267 @@
+ADReal
+INSFVTKESourceSink::computeQpResidual()
+{
+
```
```suggestion
```

grmnptr commented on modules/navier_stokes/src/fvkernels/INSFVTKESourceSink.C:
```diff
@@ -0,0 +1,267 @@
+  ADReal production = 0.0;
+  ADReal destruction = 0.0;
+  const Moose::StateArg state = determineState();
+  auto old_state =
```
```suggestion
  const auto old_state =
```

grmnptr commented on modules/navier_stokes/src/fvkernels/INSFVTKESourceSink.C:
```diff
@@ -0,0 +1,267 @@
+
+      ADReal velocity_grad_norm = parallel_speed / _dist[_current_elem][i];
+
+      /// Do not erase!!
+      /// More complete expansion for velocity gradient. Leave commented for now.
+      //// Will be useful later when doing two-phase or compressible flow
```
```suggestion
      /// Do not erase!!
      // More complete expansion for velocity gradient. Leave commented for now.
      // Will be useful later when doing two-phase or compressible flow
```

grmnptr commented on modules/navier_stokes/src/fvkernels/INSFVTKESourceSink.C:
```diff
@@ -0,0 +1,267 @@
+
+      y_plus_vec.push_back(y_plus);
+
+      ADReal velocity_grad_norm = parallel_speed / _dist[_current_elem][i];
```
```suggestion
      const ADReal velocity_grad_norm = parallel_speed / _dist[_current_elem][i];
```

grmnptr commented on modules/navier_stokes/src/fvkernels/INSFVTKESourceSink.C:
```diff
@@ -0,0 +1,267 @@
+
+    for (unsigned int i = 0; i < y_plus_vec.size(); i++)
+    {
+      auto y_plus = y_plus_vec[i];
+
+      auto fi = _face_infos[_current_elem][i];
```
```suggestion
      const auto y_plus = y_plus_vec[i];

      const auto fi = _face_infos[_current_elem][i];
```

grmnptr commented on modules/navier_stokes/src/fvkernels/INSFVTKESourceSink.C:
```diff
@@ -0,0 +1,267 @@
+      auto fi = _face_infos[_current_elem][i];
+      const bool defined_on_elem_side = _var.hasFaceSide(*fi, true);
+      const Elem * const loc_elem = defined_on_elem_side ? &fi->elem() : fi->neighborPtr();
+      Moose::FaceArg facearg = {
+          fi, Moose::FV::LimiterType::CentralDifference, false, false, loc_elem};
+      ADReal wall_mut = _mu_t(facearg, state);
```
```suggestion
      const Moose::FaceArg facearg = {
          fi, Moose::FV::LimiterType::CentralDifference, false, false, loc_elem};
      const ADReal wall_mut = _mu_t(facearg, state);
```

grmnptr commented on modules/navier_stokes/src/fvkernels/INSFVTKESourceSink.C:
```diff
@@ -0,0 +1,267 @@
+
+    const auto & grad_u = _u_var->adGradSln(_current_elem, state);
+    auto Sij_xx = 2.0 * grad_u(0);
+    ADReal Sij_xy = 0.0;
+    ADReal Sij_xz = 0.0;
+    ADReal Sij_yy = 0.0;
+    ADReal Sij_yz = 0.0;
+    ADReal Sij_zz = 0.0;
```
Same comment as before, would be much nicer if you used ADTensorValue here. You have used these in the MomentumDiffusion complete expansion I think.

tanoret replied:
I tried and it is slowing down runtime. Will revisit later in the future.

grmnptr commented on modules/navier_stokes/src/fvkernels/INSFVTKESourceSink.C:
```diff
@@ -0,0 +1,267 @@
+
+    // Production limiter - not needed for most applications
+    if (_max_mixing_length < 1e10)
+    {
+      if (std::pow(std::abs(production), 1.5) / std::abs(destruction) > _max_mixing_length)
+        production = std::pow(_max_mixing_length * std::abs(destruction.value()) + 1e-10, 2. / 3.);
+    }
```
```suggestion
      if (std::pow(std::abs(production), 1.5) / std::abs(destruction) > _max_mixing_length)
        production = std::pow(_max_mixing_length * std::abs(destruction.value()) + 1e-10, 2. / 3.);
```

grmnptr commented on modules/navier_stokes/src/fvkernels/INSFVTurbulentAdvection.C:
```diff
@@ -0,0 +1,147 @@
+
+  if (_face_type == FaceInfo::VarFaceNeighbors::ELEM ||
+      _face_type == FaceInfo::VarFaceNeighbors::BOTH)
+  {
+    if (!bounded_elem)
+    {
+      // residual contribution of this kernel to the elem element
+      prepareVectorTag(_assembly, _var.number());
+      _local_re(0) = r;
+      accumulateTaggedLocalResidual();
+    }
+  }
```
```suggestion
    if (!bounded_elem)
    {
      // residual contribution of this kernel to the elem element
      prepareVectorTag(_assembly, _var.number());
      _local_re(0) = r;
      accumulateTaggedLocalResidual();
  }
```

grmnptr commented on modules/navier_stokes/src/fvkernels/INSFVTurbulentAdvection.C:
```diff
@@ -0,0 +1,147 @@
+  }
+  if (_face_type == FaceInfo::VarFaceNeighbors::NEIGHBOR ||
+      _face_type == FaceInfo::VarFaceNeighbors::BOTH)
+  {
+    if (!bounded_neigh)
+    {
+      // residual contribution of this kernel to the neighbor element
+      prepareVectorTagNeighbor(_assembly, _var.number());
+      _local_re(0) = -r;
+      accumulateTaggedLocalResidual();
+    }
+  }
```
```suggestion
    if (!bounded_neigh)
    {
      // residual contribution of this kernel to the neighbor element
      prepareVectorTagNeighbor(_assembly, _var.number());
      _local_re(0) = -r;
      accumulateTaggedLocalResidual();
    }
```

grmnptr commented on modules/navier_stokes/src/fvkernels/INSFVTurbulentAdvection.C:
```diff
@@ -0,0 +1,147 @@
+}
+
+void
+INSFVTurbulentAdvection::computeJacobian(const FaceInfo & fi)
```
Sad that we need to override these functions, but I am supportive of this for now.

grmnptr commented on modules/navier_stokes/src/fvkernels/INSFVTurbulentDiffusion.C:
```diff
@@ -0,0 +1,186 @@
+INSFVTurbulentDiffusion::initialSetup()
+{
+  FVDiffusion::initialSetup();
+  for (const auto & elem : _fe_problem.mesh().getMesh().element_ptr_range())
+  {
+    auto wall_bounded = false;
+    for (unsigned int i_side = 0; i_side < elem->n_sides(); ++i_side)
+    {
+      const std::vector<BoundaryID> side_bnds = _subproblem.mesh().getBoundaryIDs(elem, i_side);
+      for (const BoundaryName & name : _wall_boundary_names)
+      {
+        BoundaryID wall_id = _subproblem.mesh().getBoundaryID(name);
+        for (BoundaryID side_id : side_bnds)
+        {
+          if (side_id == wall_id)
+            wall_bounded = true;
+        }
+      }
+    }
+    _wall_bounded[elem] = wall_bounded;
+  }
```
Don't you have a function for this already in `NavierStokesMethods`?

tanoret replied:
Changed. Thanks

grmnptr commented on modules/navier_stokes/src/fvkernels/INSFVTurbulentDiffusion.C:
```diff
@@ -0,0 +1,186 @@
+  using namespace Moose::FV;
+  const auto state = determineState();
+
+  auto dudn = gradUDotNormal(state);
```
```suggestion
  const auto dudn = gradUDotNormal(state);
```

grmnptr commented on modules/navier_stokes/src/materials/INSFVkEpsilonViscosityMaterial.C:
```diff
@@ -0,0 +1,42 @@
+  InputParameters params = FunctorMaterial::validParams();
+  params.addClassDescription("Computes the turbulent dynamic viscosity given k and epsilon.");
+  params.addRequiredParam<MooseFunctorName>(NS::density,
+                                            "The liquid density (not the mixture one).");
```
```suggestion
                                            "The liquid density.");
```

grmnptr reviewed (changes requested):
Done with the source files. Don't be scared, most of the comments are correction suggestions for typos. Will continue with the test input files tomorrow.

tanoret commented on modules/navier_stokes/src/base/NavierStokesMethods.C:
```diff
@@ -106,4 +136,156 @@ computeSpeed(const ADRealVectorValue & velocity)
 }
+
+/// Bounded element maps for wall treatement
+std::map<const Elem *, bool>
```
I don't really like passing the map to populate as an argument of the function. I understand the mechanics but it is quite complicated to understand when reading it. The cleaner way would be using global variables like I was doing before but this is not allowed from what I understood. I will think more about this one

tanoret replied:
Added as an argument to a function with hopefully clear enough comments

tanoret commented on modules/navier_stokes/src/base/NavierStokesMethods.C:
```diff
@@ -106,4 +136,156 @@ computeSpeed(const ADRealVectorValue & velocity)
+}
+
+/// Bounded element face distances for wall treatement
+std::map<const Elem *, std::vector<Real>>
```
Again, I don't really like to pass the map as an argument of the function. The code should be immediate to understand to a reader. For instance, when calling y = f(x), it is clear that x is the input and y the output. This is easy to read and understand. When we call void f(x,y) it is not easy to understand what is going on except if I dig into the function. Also, the cost of copying the map should be minimal wrt the cost of populating it

tanoret commented on modules/navier_stokes/src/base/NavierStokesMethods.C:
```diff
@@ -106,4 +136,156 @@ computeSpeed(const ADRealVectorValue & velocity)
+        BoundaryID wall_id = _subproblem.mesh().getBoundaryID(name);
+        for (BoundaryID side_id : side_bnds)
+        {
+          if (side_id == wall_id)
+          {
+            const FaceInfo * fi = _subproblem.mesh().faceInfo(elem, i_side);
+            _face_infos[elem].push_back(fi);
+          }
```
This is not right, is it?  _face_infos[elem] will always get populated this way

tanoret commented on modules/navier_stokes/src/fvbcs/INSFVTKEDWallFunctionBC.C:
```diff
@@ -0,0 +1,107 @@
+  if (y_plus <= 5.0) // sub-laminar layer
+  {
+    const auto laminar_value =
+        2.0 * weight * TKE * _mu(makeElemArg(&_current_elem), state) / std::pow(dist, 2);
```
It is the standard treatment for cells with multiple wall boundaries

tanoret replied:
I added an explanation on top when computing the weights

Commit:
adding von_karman and E to constants Refs #9007

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1934008/) on 1bec5d2 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 4b3cebd1fd3867810d5f9f10ada32180250bab17`

Commit:
changing refs to wall_boudnar_names Refs #9007

Commit:
ammending k eps visc #Refs 9007

tanoret commented on modules/navier_stokes/src/auxkernels/kEpsilonViscosityAux.C:
```diff
@@ -0,0 +1,219 @@
+
+  const ADReal a_c = 1 / _von_karman;
+  const ADReal b_c = 1 / _von_karman * (std::log(_E * dist / mu) + 1.0);
+  const ADReal c_c = u;
```
b_c is not a Real

tanoret commented on modules/navier_stokes/src/auxkernels/kEpsilonViscosityAux.C:
```diff
@@ -0,0 +1,219 @@
+      u_tau = new_u_tau;
+    }
+
+    mooseWarning("Could not find the wall friction velocity (mu: ",
```
This wouldn't work here. This is an internal class solve as does not necessarily needs to converge if used for the laminar sublayer. The following error is produced when trying to implement it:


/home/tanome/moose/framework/build/header_symlinks/SolutionInvalidInterface.h:23:44: error: 'class kEpsilonViscosityAux' has no member named 'registerInvalidSolutionInternal'
     static const auto __invalid_id = this->registerInvalidSolutionInternal(message);

tanoret commented on modules/navier_stokes/src/auxkernels/kEpsilonViscosityAux.C:
```diff
@@ -0,0 +1,219 @@
+      mu_t_wall = wall_val.value();
+    }
+
+    mu_t = std::max(mu_t_wall, 1e-10);
```
Thanks. Added a note.

Commit:
fixing k eps viscosity x2 Refs #9007

tanoret commented on modules/navier_stokes/src/base/NavierStokesMethods.C:
```diff
@@ -97,6 +97,36 @@ findUStar(const ADReal & mu, const ADReal & rho, const ADReal & u, const Real di
+  // Fixed point iteration method to find y_plus
+  // It should take 3 or 4 iterations
+  constexpr int MAX_ITERS{10};
+  constexpr Real REL_TOLERANCE{1e-2};
```
Yes, it is a standard for the convergence of yplus. It has to do with the accuracy to which the friction velocity is calibrated. It is better if people don't play around with this.

Commit:
changing wall function addressing functions Refs #9007

tanoret commented on modules/navier_stokes/src/base/NavierStokesMethods.C:
```diff
@@ -106,4 +136,156 @@ computeSpeed(const ADRealVectorValue & velocity)
+          }
+        }
+      }
+    }
```
This one is not right

tanoret commented on modules/navier_stokes/src/executioners/SIMPLE.C:
```diff
@@ -1136,10 +1257,18 @@ SIMPLE::execute()
     while (iteration_counter < _num_iterations && !converged(ns_residuals, ns_abs_tols))
     {
+      // Resdiual index
+      size_t residual_index = 0;
```
We can't. Residual index gets updated as we solve each system

Commit:
correcting simple.c Refs #9007

Commit:
fixing inlet bcs Refs #9007

tanoret commented on modules/navier_stokes/src/fvbcs/INSFVMixingLengthTKEDBC.C:
```diff
@@ -0,0 +1,41 @@
+InputParameters
+INSFVMixingLengthTKEDBC::validParams()
+{
+  InputParameters params = FVDirichletBCBase::validParams();
```
Thanks Added

tanoret commented on modules/navier_stokes/src/fvbcs/INSFVInletIntensityTKEBC.C:
```diff
@@ -0,0 +1,56 @@
+InputParameters
+INSFVInletIntensityTKEBC::validParams()
+{
+  InputParameters params = FVDirichletBCBase::validParams();
```
Thanks added

Commit:
fixing tked wall function Refs #9007

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1935726/) on 5ef8bb0 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format a95084cddbea271bc93671fce110873b2c6423d6`

Commit:
correcting wall temperature bc Refs #9007

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1935824/) on 285e9aa wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format a95084cddbea271bc93671fce110873b2c6423d6`

Commit:
correcting viscosity wall Refs #9007

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1935881/) on ab343e0 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format a95084cddbea271bc93671fce110873b2c6423d6`

Commit:
fixing INSFVMomentumDiffusion Refs #9007

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1935930/) on 7e97526 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format a95084cddbea271bc93671fce110873b2c6423d6`

tanoret commented on modules/navier_stokes/src/fvkernels/INSFVTKEDSourceSink.C:
```diff
@@ -0,0 +1,253 @@
+    ADReal Sij_zz = 0.0;
+
+    auto grad_xx = grad_u(0);
+    ADReal grad_xy = 0.0;
+    ADReal grad_xz = 0.0;
+    ADReal grad_yx = 0.0;
+    ADReal grad_yy = 0.0;
+    ADReal grad_yz = 0.0;
+    ADReal grad_zx = 0.0;
+    ADReal grad_zy = 0.0;
+    ADReal grad_zz = 0.0;
```
It deteriorates runtime. Should revisit later in the future

Commit:
fixing TKED source and sink kernel Refs #9007

Commit:
solving tke source and sink Refs #9007

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1935979/) on 83a80b9 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format a95084cddbea271bc93671fce110873b2c6423d6`

Commit:
correct insfv diffusion and advection Refs #9007

Commit:
correcting k epsilon material Refs #9007

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1936034/) on 07c7d97 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format a95084cddbea271bc93671fce110873b2c6423d6`

Commit:
fixing clang Refs #9007

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1936083/) on d6fa9e2 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format a95084cddbea271bc93671fce110873b2c6423d6`

grmnptr commented on modules/navier_stokes/doc/content/source/fvbcs/INSFVTurbulentViscosityWallFunction.md:
```diff
@@ -0,0 +1,149 @@
+near-wall errors. The method is designed to be used in conjunction
+with porous media treatement and not necessairly for free flow.
+
+## Non-equilibrium wall functions*
```
```suggestion
## Non-equilibrium wall functions
```

grmnptr commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKEDSourceSink.md:
```diff
@@ -0,0 +1,114 @@
+# INSFVTKEDSourceSink
+
+!syntax description /FVKernels/INSFVTKEDSourceSink
+
```
```suggestion
```

grmnptr commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKEDSourceSink.md:
```diff
@@ -0,0 +1,114 @@
+
+## Wall formulation:
+
+All cells in contact with a boundary identified in the `walls` list are applied a different
```
```suggestion
All cells in contact with a boundary identified in the [!param](/FVKernels/INSFVTKEDSourceSink/walls) list are applied a different
```

grmnptr commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKEDSourceSink.md:
```diff
@@ -0,0 +1,114 @@
+A separate formulation is used for the `sub-laminar` and `logarithmic` boundary layers.
+The determination of whether the near-wall cell lies in the laminar or logarithmic region
+is performed via the non-dimensional wall distance $y^+$.
+The non-dimensional wall distance ca be as defined differently according to the
```
```suggestion
The non-dimensional wall distance can be as defined differently according to the
```

grmnptr commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKEDSourceSink.md:
```diff
@@ -0,0 +1,114 @@
+where:
+
+- $\rho$ is the density,
+- $y_p$ is the distance to the wall to the centroid of the next-to-wall cell,
```
```suggestion
- $y_p$ is the distance from the wall to the centroid of the next-to-wall cell,
```

grmnptr commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKEDSourceSink.md:
```diff
@@ -0,0 +1,114 @@
+is specified in his kernel.
+
+!alert note
+When using near-wall treatment, we assume that the $\mu_t$ funtor is computed by an object
```
```suggestion
When using near-wall treatment, we assume that the $\mu_t$ functor is computed by an object
```

grmnptr commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKESourceSink.md:
```diff
@@ -0,0 +1,110 @@
+# INSFVTKESourceSink
+
+!syntax description /FVKernels/INSFVTKESourceSink
+
```
```suggestion
```

grmnptr commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKESourceSink.md:
```diff
@@ -0,0 +1,110 @@
+
+## Wall formulation:
+
+All cells in contact with a boundary identified in the `walls` list are applied a different
```
```suggestion
All cells in contact with a boundary identified in the [!param](/FVKernels/INSFVTKESourceSink/walls) list are applied a different
```

grmnptr commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKESourceSink.md:
```diff
@@ -0,0 +1,110 @@
+
+- $k_{old}$ is the value of the turbulent kinetic energy in the previous iteration.
+
+Whereas the linear or nonlinear formulation is used is controled by the
```
```suggestion
Whether the linear or nonlinear formulation is used can be controlled by the
```

grmnptr commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTKESourceSink.md:
```diff
@@ -0,0 +1,110 @@
+Whereas the linear or nonlinear formulation is used is controled by the
+[!param](/FVKernels/INSFVTKESourceSink/linearized_model) parameter.
+
+For the destruction formulation is different for the `sub-laminar` and `logarithmic` layers.
```
```suggestion
For the destruction, formulation is different for the `sub-laminar` and `logarithmic` layers.
```

grmnptr commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVTurbulentDiffusion.md:
```diff
@@ -0,0 +1,17 @@
+the [!param](/FVKernels/INSFVTurbulentDiffusion/walls) list.
+For any element that is in contact with a boundary identified
+in the [!param](/FVKernels/INSFVTurbulentDiffusion/walls) list,
+diffusion will be skipped for that element over all faces.
```
```suggestion
diffusion contributions will be skipped for that element over all faces.
```

grmnptr commented on modules/navier_stokes/doc/content/source/materials/INSFVkEpsilonViscosityMaterial.md:
```diff
@@ -0,0 +1,22 @@
+# INSFVkEpsilonViscosityMaterial
+
+!syntax description /Materials/INSFVkEpsilonViscosityMaterial
+
```
```suggestion
```

grmnptr reviewed (changes requested):
Posting through the documentation. Mostly typos.

tanoret committed:
Apply suggestions from code review Refs #9007

Co-authored-by: Peter German <31662443+grmnptr@users.noreply.github.com>

grmnptr reviewed (changes requested):
Okay, these are the comments for the deader files. Only 1 meaningful thing.

grmnptr commented on modules/navier_stokes/include/base/NS.h:
```diff
@@ -153,6 +153,16 @@ static const std::string C = "C";
+
+// Turbulence constants
+static constexpr Real von_karman_constant = 0.4187;
+static constexpr Real E_turb_constant = 0.4187;
```
Check if this is correct.

tanoret committed:
Apply suggestions from code review Refs #9007

Co-authored-by: Peter German <31662443+grmnptr@users.noreply.github.com>

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1949092/) on 6299303 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format ea8d25e2a7bffa624e7c3ac6f6d92d6b60503aa4`

grmnptr reviewed (changes requested):
Almost there!

tanoret commented on modules/navier_stokes/include/base/NS.h:
```diff
@@ -153,6 +153,16 @@ static const std::string C = "C";
+
+// Turbulence variables
+static const std::string TKE = "k";
+static const std::string TKED = "epsilon";
```
We talked about this in person, resolving this comment.

tanoret replied:
I added NS::mu_t

Commit:
solving normals, blockIDs, and regolding Refs #9007

tanoret commented on modules/navier_stokes/src/base/NavierStokesMethods.C:
```diff
@@ -106,4 +128,114 @@ computeSpeed(const ADRealVectorValue & velocity)
   return isZero(velocity) ? 1e-42 : velocity.norm();
 }
+
+/// Bounded element maps for wall treatement
```
Thanks. Added blockID restriction

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1949636/) on 1164ca5 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format ea8d25e2a7bffa624e7c3ac6f6d92d6b60503aa4`

tanoret commented on modules/navier_stokes/include/fvkernels/INSFVTKEDSourceSink.h:
```diff
@@ -0,0 +1,88 @@
+
+  /// Map for the previous nonlienar iterate
+  std::map<const Elem *, Real> _pevious_nl_sol;
+
```
Thanks. Remove _normal map and using only faceInfo

Commit:
adding NS::mu_t Refs #9007

tanoret commented on modules/navier_stokes/src/auxkernels/kEpsilonViscosityAux.C:
```diff
@@ -0,0 +1,227 @@
+
+  const Real a_c = 1 / NS::von_karman_constant;
+  const ADReal b_c =
+      1 / NS::von_karman_constant * (std::log(NS::E_turb_constant * dist / mu) + 1.0);
```
Accidental re-golding. Fold back to old results. Thanks

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1949685/) on 9aeb8dc wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format ea8d25e2a7bffa624e7c3ac6f6d92d6b60503aa4`

tanoret commented on modules/navier_stokes/src/auxkernels/kEpsilonViscosityAux.C:
```diff
@@ -0,0 +1,227 @@
+  params.addRequiredParam<MooseFunctorName>(NS::TKED,
+                                            "Coupled turbulent kinetic energy dissipation rate.");
+  params.addRequiredParam<MooseFunctorName>(NS::density, "Density");
+  params.addRequiredParam<MooseFunctorName>("mu", "Dynamic viscosity.");
```
Repeated

tanoret committed:
Apply suggestions from code review first batch Refs #9007

Co-authored-by: Peter German <31662443+grmnptr@users.noreply.github.com>

tanoret commented on modules/navier_stokes/src/auxkernels/kEpsilonViscosityAux.C:
```diff
@@ -0,0 +1,227 @@
+    _k(getFunctor<ADReal>(NS::TKE)),
+    _epsilon(getFunctor<ADReal>(NS::TKED)),
+    _rho(getFunctor<ADReal>(NS::density)),
+    _mu(getFunctor<ADReal>("mu")),
```
Repeated

tanoret committed:
Apply suggestions from code review second batch Refs #9007"

Co-authored-by: Peter German <31662443+grmnptr@users.noreply.github.com>

tanoret commented on modules/navier_stokes/src/auxkernels/kEpsilonViscosityAux.C:
```diff
@@ -0,0 +1,227 @@
+
+  /// Setting up convenient parameters
+  const auto state = determineState();
+  auto rho = _rho(makeElemArg(_current_elem), state);
+  auto mu = _mu(makeElemArg(_current_elem), state);
+  auto nu = mu / rho;
```
Repeated

tanoret committed:
Apply suggestions from code review third batch Refs #9007

Co-authored-by: Peter German <31662443+grmnptr@users.noreply.github.com>

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1949854/) on e00ac9a wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format ea8d25e2a7bffa624e7c3ac6f6d92d6b60503aa4`

tanoret committed:
Apply suggestions from code review fourth batch Refs #9007

Co-authored-by: Peter German <31662443+grmnptr@users.noreply.github.com>

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1949903/) on 8a2ae49 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format ea8d25e2a7bffa624e7c3ac6f6d92d6b60503aa4`

Commit:
correcting internsity -> intensity Refs #9007

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1949952/) on 6f53a32 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format ea8d25e2a7bffa624e7c3ac6f6d92d6b60503aa4`

tanoret commented on modules/navier_stokes/src/fvkernels/INSFVTKEDSourceSink.C:
```diff
@@ -0,0 +1,241 @@
+  return params;
+}
+
+INSFVTKEDSourceSink::INSFVTKEDSourceSink(const InputParameters & params)
```
Yes. It can. They can be used with Newton. We have a test for that. The issue is that it is hard to converge

tanoret replied:
The test is: `lid_driven_turb_std_wall_nonlinear`

Commit:
fixing tests Refs #9007

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1950066/) on de6510c wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format ea8d25e2a7bffa624e7c3ac6f6d92d6b60503aa4`

Commit:
solving tests x2 Refs #9007

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1950171/) on f8a1ba2 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format ea8d25e2a7bffa624e7c3ac6f6d92d6b60503aa4`

tanoret commented on modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusion.C:
```diff
@@ -30,28 +33,60 @@ INSFVMomentumDiffusion::validParams()
 
   if (onBoundary(*_face_info))
-    face_mu = _mu(makeCDFace(*_face_info), state);
+    face_mu = std::max(_mu(makeCDFace(*_face_info), state), 0.0);
```
We still should not accept negative viscosity values. This will destroy our convergence. What do you think of my change?

grmnptr replied:
I understand what the purpose of this change is. I will give the user the option to choose if they want to limit or not. I don't think this should be default. Not because I think that it is fine to have negative viscosity. I just don't like limiting things without the user's awareness. People can turn this limiting off and on. If we want to discard solutions which were obtained with negative viscosity values, we have to use the solution invalidity interface. Also, this should not be a problem if the turbulent k/epsilon are limited to be in the (0,inf) range. Negative values can come from the extrapolation for sure, especially if the gradient is large close to the boundary. You can disable the two-term expansion if that is the issue. But for now I'll let the user decide if we want to limit things during the iteration.

Commit:
correcting capping viscosity Refs #9007

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1950269/) on 304fe40 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/25752/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format ea8d25e2a7bffa624e7c3ac6f6d92d6b60503aa4`

tanoret commented on modules/navier_stokes/src/auxkernels/kEpsilonViscosityAux.C:
```diff
@@ -0,0 +1,227 @@
+
+  // Computing wall value for near-wall elements if bulk wall treatement is activated
+  // bulk_wall_treatement should only be used for very coarse mesh problems
+  if (wall_bounded && _bulk_wall_treatment)
```
Thanks. Added one test.

tanoret replied:
The added test name is: `lid_driven_turb_std_wall_bulk`

tanoret commented on modules/navier_stokes/src/auxkernels/kEpsilonViscosityAux.C:
```diff
@@ -0,0 +1,227 @@
+    if (y_plus <= 5.0) // sub-laminar layer
+      mu_t_wall = 0.0;
+    else if (y_plus >= 30.0) // log-layer
+    {
+      const auto wall_val =
+          _rho(current_argument, state) * u_tau * min_wall_dist /
+              (1 / NS::von_karman_constant * std::log(NS::E_turb_constant * y_plus)) -
+          _mu(current_argument, state);
+      mu_t_wall = wall_val.value();
+    }
+    else // buffer layer
+    {
+      const auto wall_val_log =
+          _rho(current_argument, state) * u_tau * min_wall_dist /
+              (1 / NS::von_karman_constant * std::log(NS::E_turb_constant * y_plus)) -
+          _mu(current_argument, state);
+      const auto blending_function = (y_plus - 5.0) / 25.0;
+      const auto wall_val = blending_function * wall_val_log;
+      mu_t_wall = wall_val.value();
+    }
```
Thanks. If we are in the log layer, we don't really need to compute the `blending_function`. It saves a few operations. Also, at some point, we will add a few more blending functions beyond this linear interpolation. So, I prefer to keep the buffer layer separated for it to be a bit more clear. Thanks for the comment though

tanoret commented on modules/navier_stokes/src/fvbcs/INSFVTKEDWallFunctionBC.C:
```diff
@@ -0,0 +1,103 @@
+    const auto laminar_value = 2.0 * weight * TKE * mu / std::pow(dist, 2);
+    return laminar_value.value();
+  }
+  else if (y_plus >= 30.0) // log-layer
+  {
+    const auto turbulent_value = weight * _C_mu(makeElemArg(&_current_elem), state) *
+                                 std::pow(std::abs(TKE), 1.5) /
+                                 (_mu_t(makeElemArg(&_current_elem), state) * dist);
+    return turbulent_value.value();
+  }
```
Thanks, added test `lid_driven_turb_no_wall_log_first_layer` to solve this issue

tanoret commented on modules/navier_stokes/src/fvbcs/INSFVTurbulentTemperatureWallFunction.C:
```diff
@@ -0,0 +1,126 @@
+
+  // Computing friction velocity
+  ADReal u_tau;
+  if (_linearized_yplus)
+  {
+    const ADReal a_c = 1 / NS::von_karman_constant;
+    const ADReal b_c =
+        1 / NS::von_karman_constant * (std::log(NS::E_turb_constant * wall_dist / mu) + 1.0);
+    const ADReal c_c = parallel_speed;
+    u_tau = (-b_c + std::sqrt(std::pow(b_c, 2) + 4.0 * a_c * c_c)) / (2.0 * a_c);
+  }
```
Thanks. Added test `lid_driven_turb_energy_wall_linear` to solve this issue

tanoret commented on modules/navier_stokes/src/fvbcs/INSFVTurbulentTemperatureWallFunction.C:
```diff
@@ -0,0 +1,126 @@
+  {
+    alpha = kappa / (rho * cp);
+  }
+  else if (y_plus >= 30.0) // log-layer
+  {
+    const auto Pr = cp * mu / kappa;
+    const auto Pr_ratio = Pr / _Pr_t(current_argument, state);
+    const auto jayatilleke_P =
+        9.24 * (std::pow(Pr_ratio, 0.75) - 1.0) * (1.0 + 0.28 * std::exp(-0.007 * Pr_ratio));
+    const auto wall_scaling =
+        1.0 / NS::von_karman_constant * std::log(NS::E_turb_constant * y_plus) + jayatilleke_P;
+    alpha = u_tau * wall_dist / (_Pr_t(current_argument, state) * wall_scaling);
+  }
```
Thanks. Added test `lid_driven_turb_energy_wall_log` to solve this issue

Commit:
adding tests to correct coverage Refs #9007

tanoret commented on modules/navier_stokes/src/auxkernels/kEpsilonViscosityAux.C:
```diff
@@ -0,0 +1,227 @@
+{
+  if (!_wall_boundary_names.empty())
+  {
+    NS::getWallBoundedElements(_wall_boundary_names, _c_fe_problem, _subproblem, _wall_bounded);
```
Thanks. Performed the change

tanoret commented on modules/navier_stokes/src/base/NavierStokesMethods.C:
```diff
@@ -106,4 +128,114 @@ computeSpeed(const ADRealVectorValue & velocity)
+          }
+      }
+    }
+}
```
Thank you. This was removed.

tanoret commented on modules/navier_stokes/src/fvkernels/INSFVTKESourceSink.C:
```diff
@@ -0,0 +1,261 @@
+INSFVTKESourceSink::initialSetup()
+{
+  NS::getWallBoundedElements(_wall_boundary_names, _fe_problem, _subproblem, _wall_bounded);
+  NS::getElementFaceNormal(_wall_boundary_names, _fe_problem, _subproblem, _normal);
```
Thank you. This was removed

tanoret commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/BFS_ERCOFTAC.i:
```diff
@@ -0,0 +1,397 @@
+[GlobalParams]
+  rhie_chow_user_object = 'rc'
+  advected_interp_method = ${advected_interp_method}
+  velocity_interp_method = ${velocity_interp_method}
```
Thank you. Made 'rc' the default for the velocity interpolation method, i.e., `velocity_interp_method = 'rc'`, for all inputs

tanoret commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/BFS_ERCOFTAC.i:
```diff
@@ -0,0 +1,397 @@
+  rhie_chow_user_object = 'rc'
+  advected_interp_method = ${advected_interp_method}
+  velocity_interp_method = ${velocity_interp_method}
+  two_term_boundary_expansion = true
```
Thank you. I removed `two_term_boundary_expansion=true` from global parameters

tanoret commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/lid-driven-turb-capped.i:
```diff
@@ -0,0 +1,359 @@
+    order = CONSTANT
+    family = MONOMIAL
+    fv = true
+  []
```
Removed from every input. Thank you

tanoret commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/lid-driven-turb-energy-wall.i:
```diff
@@ -0,0 +1,436 @@
+    order = CONSTANT
+    family = MONOMIAL
+    fv = true
+  []
```
Removed from every input. Thank you

tanoret commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/tests:
```diff
@@ -0,0 +1,88 @@
+    exodiff = 'lid-driven-turb-energy-wall_out.e'
+    requirement = 'The system shall be able to solve turbulence in enclosures temperature wall functions.'
+    abs_zero = 1e-6
+  []
```
As per out talk, we decided to keep a few complete input files to showcase the usage of the turbulence modeling options for external users. If this issue is becoming problematic in the future for code maintenance, I am good to change this tests to cli_args tests

tanoret commented on modules/navier_stokes/test/tests/finite_volume/ins/turbulence/tests:
```diff
@@ -0,0 +1,88 @@
+[Tests]
+  design = 'INSFVTurbulentAdvection.md INSFVTurbulentDiffusion.md INSFVTKESourceSink.md INSFVTKEDSourceSink.md INSFVTurbulentViscosityWallFunction.md kEpsilonViscosityAux.md'
+  issues = '#9007'
+  [lid_driven_turb_std_wall]
+    type = 'Exodiff'
+    input = 'lid-driven-turb-std-wall.i'
+    exodiff = 'lid-driven-turb-std-wall_out.e'
+    requirement = 'The system shall be able to solve turbulence in enclosures with standard wall functions.'
+    abs_zero = 1e-6
+  []
```
As per out talk, we decided to keep a few complete input files to showcase the usage of the turbulence modeling options for external users. If this issue is becoming problematic in the future for code maintenance, I am good to change this tests to cli_args tests

Commit:
clang corrections Refs #9007

lindsayad commented:
@GiudGiud can either @grmnptr or I take over the assignment with you in France?

GiudGiud commented:
For sure

GiudGiud commented:
For continuity, the test cases in this PR are too big and ought to be moved to the flow test bed. They are benchmarks and it s not desirable to host unconverged solutions for testing purposes.

grmnptr commented:
Would it be okay to have 2 cases of the same test: one which would only take few iterations to check if it gives the same results and another which converges to the right result (this would be heavy though). I'd be fine moving these to the flow test bed but we don't have testing for those cases yet. I would not like to delay this PR for that reason. Also, some of these cases cannot be simplified in terms of mesh resolution because the mesh needs to be a certain size to trigger different wall-treatments.

Commit:
adding short test cases Refs #9007

grmnptr commented:
The reason why these tests take a long time is because of the number of iterations needed for convergence.

GiudGiud commented:
Heavy testing is not for benchmarking either. 

Ideally the tests isolate one feature of the code, so if they start diffing we know exactly what happens. Benchmarks have their own purposes.

If you make it work, happy to see something work.

lindsayad commented on modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusion.C:
```diff
@@ -30,28 +33,66 @@ INSFVMomentumDiffusion::validParams()
+      "complete_expansion",
+      false,
+      "Boolean parameter to use complete momentum expansion is the diffusion term.");
+  params.addCoupledVar("u", "The velocity in the x direction.");
+  params.addCoupledVar("v", "The velocity in the y direction.");
+  params.addCoupledVar("w", "The velocity in the z direction.");
```
Add these as functors. More flexible and by adding as functors you won't trigger the "legacy" FV quadrature point evaluations that you get through the `Coupleable` interface

tanoret replied:
Thanks. Changed to functors

lindsayad commented on modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusion.C:
```diff
@@ -30,28 +33,66 @@ INSFVMomentumDiffusion::validParams()
+    _mu_interp_method(
+        Moose::FV::selectInterpolationMethod(getParam<MooseEnum>("mu_interp_method"))),
+    _u_var(params.isParamValid("u")
+               ? dynamic_cast<const INSFVVelocityVariable *>(getFieldVar("u", 0))
+               : nullptr),
+    _v_var(params.isParamValid("v")
+               ? dynamic_cast<const INSFVVelocityVariable *>(getFieldVar("v", 0))
+               : nullptr),
+    _w_var(params.isParamValid("w")
+               ? dynamic_cast<const INSFVVelocityVariable *>(getFieldVar("w", 0))
```
AFAICT these dynamic casts are not needed for functionality. Storing as pointers to ordinary functors should suffice

tanoret replied:
Thanks. Changed to general functors

lindsayad commented on modules/navier_stokes/src/fvkernels/INSFVTKEDSourceSink.C:
```diff
@@ -0,0 +1,242 @@
+ADReal
+INSFVTKEDSourceSink::computeQpResidual()
+{
+
```
```suggestion
```

tanoret replied:
Thanks. Removed

lindsayad commented on modules/navier_stokes/src/fvkernels/INSFVTKEDSourceSink.C:
```diff
@@ -0,0 +1,242 @@
+  const auto rho = _rho(makeElemArg(_current_elem), state);
+  const auto TKE = _k(makeElemArg(_current_elem), state);
+
+  if (_wall_bounded[_current_elem])
```
```suggestion
  if (_wall_bounded.find(_current_elem) != _wall_bounded.end())
```
This will help keep the size of your container down which will help with the O(log N) search times through this container

tanoret replied:
Thanks. Changed

tanoret commented:
Each test is designed to test a particular feature of the code, i.e., wall-function modeling approach, production/destruction formulation, etc. The problem of testing in "simplified" canonical cases is that numerical diffusion is too big and hinder some of the aspects of turbulence modeling that are quite sensitive to the modeling approach. In brief, turbulence modeling should not be tested in over-simplified regression tests. Otherwise, we can break things and never find out. I added two short test cases to be sure than run in a non-heavy configuration

lindsayad commented on modules/navier_stokes/src/fvkernels/INSFVTKESourceSink.C:
```diff
@@ -0,0 +1,262 @@
+      _linearized_model ? Moose::StateArg(1, Moose::SolutionIterationType::Nonlinear) : state;
+  const auto rho = _rho(makeElemArg(_current_elem), state);
+
+  if (_wall_bounded[_current_elem])
```
```suggestion
  if (_wall_bounded.find(_current_elem) != _wall_bounded.end())
```

tanoret replied:
Thanks. Changed

lindsayad commented on modules/navier_stokes/src/fvkernels/INSFVTurbulentAdvection.C:
```diff
@@ -0,0 +1,144 @@
+  auto bounded_elem = _wall_bounded[elem];
+  auto bounded_neigh = _wall_bounded[neighbor];
+
+  if (_face_type == FaceInfo::VarFaceNeighbors::ELEM ||
+      _face_type == FaceInfo::VarFaceNeighbors::BOTH)
+    if (!bounded_elem)
```
Just put this all in one `if`

tanoret replied:
Thanks. Changed

lindsayad commented on modules/navier_stokes/src/fvkernels/INSFVTurbulentAdvection.C:
```diff
@@ -0,0 +1,144 @@
+      _local_re(0) = r;
+      accumulateTaggedLocalResidual();
+    }
+  if (_face_type == FaceInfo::VarFaceNeighbors::NEIGHBOR ||
+      _face_type == FaceInfo::VarFaceNeighbors::BOTH)
+    if (!bounded_neigh)
```
Just put this all in one `if`

tanoret replied:
Thanks. Changed

lindsayad commented on modules/navier_stokes/src/fvkernels/INSFVTurbulentAdvection.C:
```diff
@@ -0,0 +1,144 @@
+
+  const Elem * elem = fi.elemPtr();
+  const Elem * neighbor = fi.neighborPtr();
+  auto bounded_elem = _wall_bounded[elem];
+  auto bounded_neigh = _wall_bounded[neighbor];
```
```suggestion
  const bool bounded_elem = _wall_bounded.find(elem) != _wall_bounded.end();
  const bool bounded_neigh = _wall_bounded.find(neighbor) != _wall_bounded.end();
```

tanoret replied:
Thanks. Changed

lindsayad commented on modules/navier_stokes/src/fvkernels/INSFVTurbulentAdvection.C:
```diff
@@ -0,0 +1,144 @@
+
+  const Elem * elem = fi.elemPtr();
+  const Elem * neighbor = fi.neighborPtr();
+  auto bounded_elem = _wall_bounded[elem];
+  auto bounded_neigh = _wall_bounded[neighbor];
```
```suggestion
  const bool bounded_elem = _wall_bounded.find(elem) != _wall_bounded.end();
  const bool bounded_neigh = _wall_bounded.find(neighbor) != _wall_bounded.end();
```

lindsayad commented on modules/navier_stokes/src/fvkernels/INSFVTurbulentDiffusion.C:
```diff
@@ -0,0 +1,166 @@
+
+  const Elem * elem = fi.elemPtr();
+  const Elem * neighor = fi.neighborPtr();
+  auto bounded_elem = _wall_bounded[elem];
+  auto bounded_neigh = _wall_bounded[neighor];
```
```suggestion
  const bool bounded_elem = _wall_bounded.find(elem) != _wall_bounded.end();
  const bool bounded_neigh = _wall_bounded.find(neighbor) != _wall_bounded.end();
```

tanoret replied:
Thanks. Changed

lindsayad commented on modules/navier_stokes/src/fvkernels/INSFVTurbulentDiffusion.C:
```diff
@@ -0,0 +1,166 @@
+  auto r = MetaPhysicL::raw_value(fi.faceArea() * fi.faceCoord() * computeQpResidual());
+
+  const Elem * elem = fi.elemPtr();
+  const Elem * neighor = fi.neighborPtr();
```
```suggestion
  const Elem * neighbor = fi.neighborPtr();
```

tanoret replied:
Thanks. Changed

lindsayad commented on modules/navier_stokes/src/fvkernels/INSFVTurbulentDiffusion.C:
```diff
@@ -0,0 +1,166 @@
+  auto bounded_elem = _wall_bounded[elem];
+  auto bounded_neigh = _wall_bounded[neighor];
+
+  if (_face_type == FaceInfo::VarFaceNeighbors::ELEM ||
+      _face_type == FaceInfo::VarFaceNeighbors::BOTH)
+    if (!bounded_elem)
```
All one `if`

tanoret replied:
Thanks. Changed

lindsayad commented on modules/navier_stokes/src/fvkernels/INSFVTurbulentDiffusion.C:
```diff
@@ -0,0 +1,166 @@
+      _local_re(0) = r;
+      accumulateTaggedLocalResidual();
+    }
+  if (_face_type == FaceInfo::VarFaceNeighbors::NEIGHBOR ||
+      _face_type == FaceInfo::VarFaceNeighbors::BOTH)
+    if (!bounded_neigh)
```
All one `if`

tanoret replied:
Thanks. Changed

lindsayad commented on modules/navier_stokes/src/fvkernels/INSFVTurbulentDiffusion.C:
```diff
@@ -0,0 +1,166 @@
+
+  const Elem * elem = fi.elemPtr();
+  const Elem * neighbor = fi.neighborPtr();
+  auto bounded_elem = _wall_bounded[elem];
+  auto bounded_neigh = _wall_bounded[neighbor];
```
```suggestion
  const bool bounded_elem = _wall_bounded.find(elem) != _wall_bounded.end();
  const bool bounded_neigh = _wall_bounded.find(neighbor) != _wall_bounded.end();
```

tanoret replied:
Thanks. Changed

Commit:
addressing alex review Refs #9007

grmnptr committed:
Transition more towards functor evaluations to decrease need qp-based evaluations. (#9007)

grmnptr committed:
Add shorter tests for non-heavy testing. (#9007)

grmnptr committed:
Alter test requirements to use details. (#9007)

moosebuild commented:
Job [Python 3.8](https://civet.inl.gov/job/1959906/) on ee31946 : invalidated by @grmnptr

Not related

lindsayad committed:
Small changes

- Iterate over active elements else seg fault
- Whitespace removal
- Typo fix
- More `auto`

lindsayad committed:
Fix some typos in INSFVMixingLengthTKEDBC.C

lindsayad committed:
Don't do repeat map lookups

And also use `NS::computeSpeed` which will allow the nonlinear solver
to work with zero initial conditions for velocity

lindsayad committed:
Typo fixes in lid driven turb std wall inputs

lindsayad committed:
Print nonlinear residuals for Newton

grmnptr committed:
Add warning for negative viscosity and modify tests so we don't encounter any. (#9007)

grmnptr reviewed (approved):
Thabks for the great contrib, @tanoret !
